worker/caasoperator/caasoperator.go

```diff
@@ -213,10 +213,15 @@ func NewWorker(config Config) (worker.Worker, error) {
 		return nil, errors.Trace(err)
 	}
 	paths := config.getPaths()
+	logger := loggo.GetLogger("juju.worker.uniter.charm")
```

Comment:
Maybe pass this 

loggo.GetLogger("juju.worker.uniter")

in as a second logger (eg uniterLogger) on the Config struct, and then use Child("charm") off that second logger and also below...

Reply:
Passing through the charm logger here keeps the logging at the same place. It is because both of the types here were from the charm package, and hence used that logger.

I do feel like we could do a cleanup pass over everything later to clean up some of these.